AppCenter/AppCenter/Internals/Util/MSPerformSelectorUtil.m

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSPerformSelectorUtil.h"
+#import "MSAppCenterInternal.h"
+
+@implementation MSPerformSelectorUtil
+
++ (void)performSelectorOnMainThread:(NSObject *)source withSelector:(SEL)selector withObjects:(NSObject *)objects, ... {
+  NSMethodSignature *signature = [source methodSignatureForSelector:selector];
+  if (!signature) {
+    MSLogError([MSAppCenter logTag], @"MSUtility: Method not found.");
+    return;
+  }
+  NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:signature];
```

Comment:
`NSInvocation` can be costly. Would it make sense to use `performSelectorOnMainThread` directly when number of parameter is less than 2. That would be an optimization.

Reply:
Actually even better can we just use blocks as usual + `performSelectorOnMainThread` directly? it's just theory but the caller would call a method that takes a block on  `MSPerformSelectorUtil`(something like: `performBlockOnMainThread:`). Then `performBlockOnMainThread:` would simply call a selector of `MSPerformSelectorUtil` itself on the main thread that would run the given block.

Reply:
If that's almost the same performance than `dispatch_async(dispatch_get_main_queue()` then we can indeed to it for all the platforms everywhere.

Reply:
Replaced the current invocation method to perform block for macOS. The current solution has been tested on ios, macos, tvos sasquatch projects (works correctly). In reachability implementation also updated tests passed.